flutter_example/lib/macros/render_accessors.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:macro_builder/definition.dart';
+
+class RenderAccessors implements FieldDeclarationMacro {
+  const RenderAccessors({
+    this.needsPaint = false,
+    this.needsLayout = false,
+    this.needsSemantics = false,
+  });
+
+  final bool needsPaint;
+  final bool needsLayout;
+  final bool needsSemantics;
+
+  @override
+  void visitFieldDeclaration(
+    FieldDeclaration declaration,
+    ClassDeclarationBuilder builder,
+  ) {
+    // TODO(goderbauer): Check that the annotated field is inside a RenderObject. How?
```

Comment:
Interesting point, I think this is something we need to enable. I will work on figuring out something that doesn't break our ordering model.

I think it should be ok to give you the previous phases information for parent declarations, without introducing ordering weirdness.

Reply:
https://github.com/jakemac53/macro_prototype/issues/21